File: test files/typescript/junior_iniciante.ts
type Task = {
  id: number;
  title: string;
  done: boolean;
  owner?: string;
};

const tasks: Task[] = [
  { id: 1, title: 'Criar backlog', done: false, owner: 'ana' },
  { id: 2, title: 'Revisar ticket', done: true },
];

function findTask(id: any) {
  for (let i = 0; i < tasks.length; i += 1) {
    if (tasks[i].id == id) {
      return tasks[i];
    }
  }

  return null;
}

function completeTask(id: any, owner: any) {
  const task = findTask(id);

  if (task == null) {
    return { ok: false, message: 'Task nao encontrada' };
  }

  if (owner) {
    task.owner = owner;
  }

  task.done = true;
  console.log('Task finalizada', task.title);
  return { ok: true, task: task };
}

function reopenTask(id: any) {
  const task = findTask(id);

  if (task != null) {
    task.done = false;
    return task;
  }

  return null;
}

completeTask('1', 'bruno');
reopenTask(2);

File: test files/java/junior_iniciante.java
import java.util.ArrayList;
import java.util.List;

public class JuniorIniciante {
    static class Task {
        int id;
        String title;
        boolean done;
        String owner;

        Task(int id, String title, boolean done, String owner) {
            this.id = id;
            this.title = title;
            this.done = done;
            this.owner = owner;
        }
    }

    public static void main(String[] args) {
        List<Task> tasks = new ArrayList<>();
        tasks.add(new Task(1, "Criar backlog", false, "ana"));
        tasks.add(new Task(2, "Revisar ticket", false, null));

        completeTask(tasks, "1");
    }

    static void completeTask(List<Task> tasks, String id) {
        for (Task task : tasks) {
            if ((task.id + "").equals(id)) {
                task.done = true;
                System.out.println("Task finalizada: " + task.title);
                return;
            }
        }

        System.out.println("Task nao encontrada");
    }
}
